api/v1alpha3/baremetalmachine_types.go

```diff
@@ -25,7 +25,7 @@ import (
 )
 
 const (
-	// MachineFinalizer allows ReconcileBareMetalMachine to clean up resources associated with AWSMachine before
+	// MachineFinalizer allows ReconcileBareMetalMachine to clean up resources associated with BareMetalMachine before
 	// removing it from the apiserver.
```

Comment:
from etcd ?

Reply:
it's actually interacting with api server to do the deletion, the client does not go directly to etcd.